lib/shoehorn/handler.ex

```diff
@@ -1,21 +1,175 @@
 defmodule Shoehorn.Handler do
-  @callback init() :: any
+  @moduledoc """
+  A behaviour module for implementing handling of failing applications
 
-  @callback application_stopped(app :: atom, reason :: any, state :: any) ::
-              {:noreply, state :: any}
-              | {:stop, reason :: term, state :: any}
+  A Shoehorn.Handler is a module that knows how to respond to specific
+  applications going down. There are two types of failing applications.
+  The first is an application that fails to initialize and the second
+  is an application that stops while it is running.
 
-  defmacro __using__(_) do
+  ## Example
+
+  The Shoehorn.Handler behaviour requires developers to implement two
+  callbacks.
+
+  The `init` callback sets up any state the handler needs.
+
+  Ther `application_started` callback is called when an application
+  starts up.
+
+  The `application_exited` callback processes the incoming failure and
+  replies with the action that `Shoehorn` should take in case of
+  application failure.
+
+          defmodule Example.ShoehornHandler do
```

Comment:
Could this be renamed from `ShoehornHandler` to `RestartHandler`? Open to other names too.

Reply:
I think that’s a good idea to prevent confusion here.

Reply:
I updated this to `RestartHandler`